src/components/v1/V1Project/modals/V1ConfirmPayOwnerModal.tsx

```diff
@@ -195,43 +229,87 @@ export default function V1ConfirmPayOwnerModal({
             name="memo"
             label={t`Memo (optional)`}
             className={'antd-no-number-handler'}
-            extra={t`Add an on-chain memo to this payment.`}
+            style={{ marginBottom: 0 }}
           >
-            <MemoFormInput />
```

Comment:
**Question:** why remove this? 

Lines `297:312` are repeated 3 times throughout the codebase I believe. I think it would be great if we could leverage the `<MemoFormInput` in both v1 and v2 pay modals to avoid as much repeated code as possible

Reply:
That is a reasonable assumption and question, i'll try to be concise and explain to you why I went this route.

`MemoFormInput` is acting as a shared component and in order to function in pay modal like form inputs are acting in v2 it would need some modifications.

Currently, it is outputting `value` and `value` is formed both from `Input.TextArea` and from `Sticker` components, therefore outputted value is a mix of text and URLs. 
The way `memo` is built in v2 components is that it is using `buildPaymentMemo` function and in the end, it is combining value based on text value, sticker URLs, and image URLs from all separate form items and passing it to `pay` function.

We are trying to achieve the same process in v1 and with the current logic from `MemoFormInput` it is not possible, however, what I can do is, I can change how `MemoFormInput` works and separate logic there, so it can output `text` and `stickerURLs` but it would require changes on all files that are using `MemoFormInput`.

That is absolutely not a problem for me, but I do not want to go far away before we are both on the same page and you understand what I am trying to achieve. 
If you find the above confusing, I'm open to jumping on a quick call and explaining everything to you ❤️

Reply:
Ok thanks for the explanation @NakedFool , too easy!